### PR TITLE
Blg-18/Api improvements (Part 2)

### LINKED ISSUE #18: Improve the backend API
- [x] Allow filtering posts.
- [x] Allow filtering comments.
- [ ] Allow filtering votes.
- [ ] Add pagination.
- [ ] Improve tests (partially done).

### PR DESCRIPTION
**Pull request by adambudziak**
Resolves #18 (partially, again; there will be a Part 3, I believe)

* Added filters to posts and comments: boolean filter for `verified` and a char filter for the author's username (filtering for an anonymous author doesn't work yet);
* Added a whole new integration tests setup with tavern and a lot of tests (it's kinda lame for now, but it works);
* Slightly cleaned up backend.

The remaining todos are listed in #18.

### CODE REVIEW

File: backend/posts/views/posts.py [hunk 1]
```diff
@@ -53,43 +56,25 @@ def perform_create(self, serializer):
...
 
 
 class UserViewSet(viewsets.ReadOnlyModelViewSet):
     queryset = User.objects.all()
     serializer_class = UserSerializer
+
+
+@api_view(['GET'])
+def post_comments_view(request, post_pk):
+    all_comments = Comment.objects.filter(post__pk=post_pk)
```
Comment (adambudziak) [inline backend/posts/views/posts.py]: I'd rather have it inside `CommentViewSet` but I don't know how to do it yet.

File: backend/posts/filters.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+from django import forms
+from django.core.exceptions import ValidationError
+from django_filters import widgets, filters, rest_framework as rest_filters
+
+
+class StrictBooleanWidget(widgets.BooleanWidget):
+    """
+    Widget with functionality similar to BooleanWidget provided with django_filters
+    but that allows to handle invalid values instead of just ignoring them.
+    """
```
Comment (adambudziak) [inline backend/posts/filters.py]: Normally, `BooleanFilter` shipped with `django_filters` uses `BooleanWidget` which converts all invalid values for a boolean filter, such as "dog", "maybe" or "fasle" into `None` thus disabling the filter.

I don't really see any advantage to keep this behavior, so I created this `StrictBooleanWidget` class that, when used with an appropriate field class (below) allows me to response with `400 Bad Request` if the user provides something else than True, False, 0 or 1 or None.

(`django_filters` has also a second implementation of `BooleanFilter` that uses `NullBooleanField` and has the same behavior that I dislike)

**Comment by adambudziak** [comment]
Circle CI will not work until we merge #24 here.

File: backend/posts/filters.py (hunk 2 above)
Comment (Mc01) [reply backend/posts/filters.py]: Seems legit for me. Thx for explanation 👍

**Pull request merged**